src/sentry/search/events/builder/metrics.py

```diff
@@ -159,7 +164,14 @@ def _get_on_demand_metric_spec(self, field: str) -> OnDemandMetricSpec | None:
 
         groupby_columns = self._get_group_bys()
 
-        if not should_use_on_demand_metrics(self.dataset, field, self.query, groupby_columns):
+        if not should_use_on_demand_metrics(
+            dataset=self.dataset,
+            aggregate=field,
+            query=self.query,
+            groupbys=groupby_columns,
+            query_builder_call=True,
+            use_next_spec_version=self.use_next_spec_version,
```

Comment:
I feel like this might backfire due to the added complexity, since we are propagating params all over the place. I would suggest to make a function `should_use_on_demand_metrics_something` specifically for query builders and inside that have a dictionary which has a `op: [feature_flag]` dictionary, signaling that some ops can only be used if the flag is enabled.

When the call to the method is made, you can early return if the op you pass is in the dictionary and the flag is false, signaling that that op is only available as on-demand if all the flags connected to it are enabled.

WDYT?

Reply:
I do agree. Let me see what I can do.